docs/user_manual.md

```diff
@@ -49,40 +60,341 @@ python -m virtualenv venv
+Roughly, such build tree can be created by the following commands (arguments
+are omitted for simplicity):
 ``` bash
-datum --help
+datum create
+
+# describe the first source
+datum add
+datum filter
+datum transform
+datum transform
+
+# describe the second source
+datum add
+datum model add
+datum transform
+datum transform
+
+# now, the resulting dataset can be built with
+datum export
 ```
 
-As a python module:
-> The directory containing Datumaro should be in the `PYTHONPATH`
-> environment variable or `cvat/datumaro/` should be the current directory.
+### Project layout
 
 ``` bash
-python -m datumaro --help
-python datumaro/ --help
-python datum.py --help
+project/
+├── .dvc/
+├── .dvcignore
+├── .git/
+├── .gitignore
+├── .datumaro/
+│   ├── cache/ # object cache
+│   │   ├── <2 leading symbols of obj hash>/
+│   │   │   └── <rest symbols of obj hash>/
+│   │   │       └── <object data>
+│   ├── models/
+│   ├── plugins/ # custom project-specific plugins
+│   │   ├── plugin1/
+│   │   |   ├── __init__.py
+│   │   |   └── file2.py
+│   │   ├── plugin2.py
+│   │   └── ...
+│   ├── tmp/ # temp files
+│   └── tree/ # working tree metadata
+│       ├── config.yml
+│       └── sources/
+│           ├── <source name 1>.dvc
+│           ├── <source name 2>.dvc
+│           └── ...
+│
+├── <source name 1>/
+│   └── <source data>
+└── <source name 2>/
+    └── <source data>
+```
+
+### Use cases
+
+Consider few diagrams describing what Datumaro does for you behind the scene.
+
+Imagine we have a project with 2 data sources. Roughly, it corresponds to
+the following set of commands:
+
+```bash
+datum create
+datum add <...> -n source1
+datum add <...> -n source2
+datum transform <...> source1 # used 3 times
+datum transform <...> source2 # used 5 times
 ```
 
-As a python library:
+Then, for some reasons, the project cache was cleaned from `source1` revisions.
+We also don't have anything in the project working directories.
+Let's see what happens, if we call the `diff` command with 2 different
+revisions now.
 
-``` python
-import datumaro
+![use case 1](images/usecase1_diag.svg)
+
+Datumaro needs to reproduce 2 dataset revisions requested so that they could
+be read and compared. `source1` will be looked for in the project cache.
+It will be found missing, since the cache was cleaned (1). Then, Datumaro
+will look for previous source revisions and won't find any (2, 3). Then, it
+will try to download the original data and reproduce the resulting dataset,
+re-applying all the operations from the source history (4, 5). When it is
+done, the resulting dataset is returned (6).
+
+The `source2` will be looked for the same way. In our case, it will be found
+in the cache and returned. Once both datasets are restored and read, they
+are compared.
+
+Consider the second situation. Let's try to `export` the `source1`. Suppose
+we have a clear project cache and the `source1` has a copy in the working
+directory.
+
+![use case 2](images/usecase2_diag.svg)
```

Comment:
1. In step 2, shouldn't "find stage" come first"? Otherwise, what are you computing the hash of?
2. "Not Found a stage" -> "Didn't find a stage"
3. The text at the bottom should be removed and integrated into the document instead.
4. The page icon in the "Working copy" column should be labeled.

Reply:
Updated the diagrams.